zhaquirks/tuya/ts0601_sensor.py

```diff
@@ -334,154 +334,59 @@ class TuyaTempHumiditySensorVar04(CustomDevice):
-            TuyaPowerConfigurationCluster2AAA.ep_attribute,
-            "battery_percentage_remaining",
-            converter=lambda x: x * 2,  # double reported percentage
-        ),
-    }
-
-    data_point_handlers = {
-        3: "_dp_2_attr_update",
-        5: "_dp_2_attr_update",
-        15: "_dp_2_attr_update",
-    }
-
+(
+    TuyaQuirkBuilder("_TZE284_aao3yzhs", "TS0601")
+    .also_applies_to("_TZE284_sgabhwa6", "TS0601")
+    .add_tuya_dp(
+        5,
+        TuyaTemperatureMeasurement.ep_attribute,
+        "measured_value",
+        converter=lambda x: x * 10,
+    )
+    .add_tuya_dp(
+        3,
+        TuyaSoilMoisture.ep_attribute,
+        "measured_value",
+        converter=lambda x: x * 100,
+    )
+    .add_tuya_dp(
+        15,
+        TuyaPowerConfigurationCluster2AAA.ep_attribute,
+        "battery_percentage_remaining",
+        converter=lambda x: x * 2,
+    )
+    .adds(TuyaPowerConfigurationCluster2AAA)
+    .adds(TuyaSoilMoisture)
+    .adds(TuyaTemperatureMeasurement)
+    .add_to_registry()
+)
 
-class TuyaSoilSensorVar02(CustomDevice):
-    """Tuya temp and humidity sensor (variation 05)."""
 
-    signature = {
-        # "profile_id": 260,
-        # "device_type": "0x0051",
-        # "in_clusters": ["0x0000","0x0004","0x0005","0xed00","0xef00"],
-        # "out_clusters": ["0x000a","0x0019"]
-        MODELS_INFO: [
-            ("_TZE284_aao3yzhs", "TS0601"),
-            ("_TZE284_sgabhwa6", "TS0601"),
-        ],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    0xED00,
-                    TemperatureHumidityManufCluster.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
-            }
-        },
-    }
-
-    replacement = {
-        SKIP_CONFIGURATION: True,
-        ENDPOINTS: {
-            1: {
-                DEVICE_TYPE: zha.DeviceType.TEMPERATURE_SENSOR,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    SoilManufClusterVar02,
-                    TuyaTemperatureMeasurement,
-                    TuyaSoilMoisture,
-                    TuyaPowerConfigurationCluster2AAA,
-                ],
-                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
-            }
-        },
-    }
+(
+    TuyaQuirkBuilder("_TZE200_myd45weu", "TS0601")
+    .also_applies_to("_TZE200_ga1maeof", "TS0601")
+    .also_applies_to("_TZE200_9cqcpkgb", "TS0601")
+    .also_applies_to("_TZE204_myd45weu", "TS0601")
+    .add_tuya_dp(
+        5,
+        TuyaTemperatureMeasurement.ep_attribute,
+        "measured_value",
+        converter=lambda x: x * 100,
```

Comment:
Hmm, although we need to treat the temperature datapoint slightly differently in these quirks (`* 10` vs `* 100`).
Maybe it does make sense to use `TuyaQuirkBuilder` for this then, since we'd need two separate temperature clusters otherwise (although one could extend the other one there and just override a value like `multiplication = 10` that's used in the base (with a default).